### PR TITLE
DIT 1051 - Enable Search for Liabilities

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala [hunk 1]
```diff
@@ -105,10 +105,11 @@ class BindingTariffClassificationConnector @Inject()(appConfig: AppConfig, clien
       search.commodityCode.map(qb.unbind("commodity_code", _)),
       search.decisionDetails.map(qb.unbind("decision_details", _)),
       search.status.map(_.map(s => qb.unbind("status", s.toString)).mkString("&")),
+      search.applicationType.map(_.map(s => qb.unbind("application_type", s.toString)).mkString("&")),
       search.keywords.map(_.map(k => qb.unbind("keyword", k)).mkString("&"))
     ).filter(_.isDefined).map(_.get)
 
-    val url = s"${appConfig.bindingTariffClassificationUrl}/cases?application_type=BTI&${(reqParams ++ optParams).mkString("&")}&page=${pagination.page}&page_size=${pagination.pageSize}"
+    val url = s"${appConfig.bindingTariffClassificationUrl}/cases?${(reqParams ++ optParams).mkString("&")}&page=${pagination.page}&page_size=${pagination.pageSize}"
```
Comment (aaroncarroll82) [inline app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala]: Wonder if it'd be worth pulling in the changes from branch for DIT-1078 and expanding/using the buildQuery method there?

File: app/uk/gov/hmrc/tariffclassificationfrontend/models/Search.scala [hunk 2]
```diff
@@ -27,11 +28,12 @@ case class Search
   commodityCode: Option[String] = None,
   decisionDetails: Option[String] = None,
   status: Option[Set[PseudoCaseStatus]] = None,
+  applicationType: Option[Set[ApplicationType]] = None,
   keywords: Option[Set[String]] = None
 ) {
 
   def isEmpty: Boolean = {
-    // Live rulings only omitted intentionally as it is a post-search filter
+    // Status & Application Type intentionally as it is a post-search filter
```
Comment (aaroncarroll82) [inline app/uk/gov/hmrc/tariffclassificationfrontend/models/Search.scala]: You've omitted the word omitted ;) IF we have comments in, they should make sense

**Comment by aaroncarroll82** [comment]
On the search results, the liabilities are not shown first in the results, should there be similar rules here as there are on the case list views, with the live liabilities top, then non live, then BTI?

File: app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala (hunk 1 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala]: Is DIT-1078 ready for merging?

Comment (aaroncarroll82) [reply app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala]: It is unless anyone else has any comments on it

**Comment by SteveSugden** [comment]
@aaroncarroll82 - having discussed the liability search results with Matt et al I think the way they are displayed / ordered is ok (i.e. doesn't need to change).   The thinking is that the search results are not indicating priority of work (i.e. don't need to bring the live liabilities to the top).

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/advanced_search.scala.html [hunk 3]
```diff
@@ -101,9 +119,10 @@ <h2 class="heading-large">Search results</h2>
                                     CheckOption("Rejected", PseudoCaseStatus.REJECTED.toString, checked = formHasStatus(PseudoCaseStatus.REJECTED)),
                                     CheckOption("Suppressed", PseudoCaseStatus.SUPPRESSED.toString, checked = formHasStatus(PseudoCaseStatus.SUPPRESSED))
                                 ),
-                                legend = Some("Filter by status"),
+                                legend = None,
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/advanced_search.scala.html]: None is the default for legend

File: app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala (hunk 1 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala]: DIT-1078 is merged in now

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala]: I'm not sure the builder adds much for this instance (I can understand why it helps with the queues though)

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/advanced_search.scala.html (hunk 3 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/advanced_search.scala.html]: Good point

**Pull request merged**